.github/workflows/pypi-package.yml

```diff
@@ -0,0 +1,67 @@
+---
+name: Build & maybe upload PyPI package
+
+on:
+  push:
+    branches: [main]
+    tags: ["*"]
+  pull_request:
+    branches: [main]
+  release:
+    types:
+      - published
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+  # Always build & lint package.
+  build-package:
+    name: Build & verify package
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - uses: hynek/build-and-inspect-python-package@v1
```

Comment:
The source is at https://github.com/hynek/build-and-inspect-python-package

I find the preview build artifacts and summaries of the package contents useful.

It's good to re-use the same packages from step 1 in steps 2/3 to avoid duplicated work, and so we know the files we may have inspected are the same. 

You can still do `python -m build` on the command line. 

This is adding extra automation for the CI, so we can have machines do extra work for humans, and the info is there for when we need it.

Reply:
> we know the files we may have inspected are the same.

I take this point, but if the release is automatic, there is no 'checking' step -- the release just happens.

I think the same challenge applies to *build-and-inspect-python-package* -- I'd like to be able to acertain that what I run locally is the same as what runs on CI, and as it's an action I can't run it locally -- would it be possible to have either that action or the commands it runs as a script file that can be run on one's computer?

Reply:
The whole point of CI/CD is we don't need to run it locally.

But if you do, it's good old `python -m build` to build the sdist and wheel.

If you want to inspect the sdist, just unzip it. For the wheel, rename to .zip and unzip it.

And then it would be twine to upload but we want to use OIDC/trusted publishing to avoid having secrets on people's machines, or long-living personal API tokens in the repo settings.

Reply:
I suppose I'd like fewer levels of abstraction, but I don't want to further block this PR, so I'll acquiesce!